### PR TITLE
[v9] Move Auth/Proxy-only Kubernetes guides

### PR DESCRIPTION
**Pull request by ptgott**
Backports #13068

See #12654

The Kubernetes Access section includes guides for using the Kubernetes
Service as well as guides for running the Auth and Proxy Service on
Kubernetes. This is misleading, since (a) you can run the Auth/Proxy
on Kubernetes without using Kubernetes Access and (b) you can use
Kubernetes Access without running the Auth/Proxy on Kubernetes.

This change focuses on the pages in:

/docs/pages/kubernetes-access/getting-started/

- Move docs/pages/kubernetes-access/getting-started guides that don't
  relate to Kubernetes Access out of this section, and change the
  docs/pages/kubernetes-access/getting-started.mdx page to point to
  the current "Agent" guide.

- Edit the "docs/pages/getting-started.mdx" menu page to accommodate the
  Kubernetes guides. Previously, this page included separate links for
  each edition's getting started guide. To add the Kubernetes Cluster
  guide, I organized the links in the "Deploy to production" section into
  a TabItem for each Teleport edition. There are separate links to the
  Kubernetes Cluster guide, one for OSS users and one for Enterprise
  users. These links also take users to the appropriate scope.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/config.json [hunk 1]
```diff
@@ -29,6 +29,15 @@
...
               "hideInScopes": ["enterprise", "cloud"]
+            },
+            {
+              "title": "Local Kubernetes Lab",
+              "slug": "/getting-started/local-kubernetes/"
+            },
+            {
+              "title": "Kubernetes Cluster",
+              "slug": "/getting-started/kubernetes-cluster/",
+              "hideInScopes": "cloud"
```
Comment (klizhentas) [inline docs/config.json:40]: @ptgott please remove `hideInScopes`

**Review by klizhentas — CHANGES_REQUESTED** [review]

**Comment by ptgott** [comment]
@klizhentas after rebasing, the changes no longer include `hideInScopes`.

**Review by stevenGravy — APPROVED** [review]

**Comment by ptgott** [comment]
@klizhentas Just following up to see if the changes you requested have now been made

**Comment by ptgott** [comment]
@klizhentas Just following on this one since you requested changes. Thanks!

**Review by klizhentas — APPROVED** [review]

**Pull request merged**